beacon-chain/p2p/custody.go

```diff
@@ -114,6 +114,38 @@ func (s *Service) UpdateCustodyInfo(earliestAvailableSlot primitives.Slot, custo
 	return earliestAvailableSlot, custodyGroupCount, nil
 }
 
+// UpdateEarliestAvailableSlot updates only the earliest available slot while preserving
+// the current custody group count.
+func (s *Service) UpdateEarliestAvailableSlot(earliestAvailableSlot primitives.Slot) error {
```

Comment:
It's OK to decrease the `earliestAvailableSlot` (for example, in case of backfill).
However, it's not ok to increase it to a value in the `MIN_EPOCHS_FOR_BLOCK_REQUESTS` period.
(Because in this case it means that we voluntary refuse to serve some mandatory data).